dask_expr/reductions.py

```diff
@@ -39,41 +68,118 @@ class ApplyConcatApply(Expr):
     combine_kwargs = {}
     aggregate_kwargs = {}
 
-    def __dask_postcompute__(self):
-        return toolz.first, ()
+    @property
+    def _meta(self):
+        meta = meta_nonempty(self.frame._meta)
+        meta = self.chunk(meta, **self.chunk_kwargs)
+        meta = self.combine([meta], **self.combine_kwargs)
+        meta = self.aggregate([meta], **self.aggregate_kwargs)
+        return make_meta(meta)
 
-    def _layer(self):
-        # Normalize functions in case not all are defined
-        chunk = self.chunk
-        chunk_kwargs = self.chunk_kwargs
+    def _divisions(self):
+        return (None, None)
 
-        if self.aggregate:
-            aggregate = self.aggregate
-            aggregate_kwargs = self.aggregate_kwargs
-        else:
-            aggregate = chunk
-            aggregate_kwargs = chunk_kwargs
+    def _simplify_down(self):
+        aca_ref = ExprReference(self)
+        chunked = Chunk(self.frame, aca_ref)
+        reduced = TreeReduction(chunked, aca_ref)
+        return reduced
 
-        if self.combine:
-            combine = self.combine
-            combine_kwargs = self.combine_kwargs
-        else:
-            combine = aggregate
-            combine_kwargs = aggregate_kwargs
 
-        d = {}
-        keys = self.frame.__dask_keys__()
+class Chunk(Blockwise):
+    """Partition-wise component of `ApplyConcatApply`
 
-        # apply chunk to every input partition
-        for i, key in enumerate(keys):
-            if chunk_kwargs:
-                d[self._name, 0, i] = (apply, chunk, [key], chunk_kwargs)
-            else:
-                d[self._name, 0, i] = (chunk, key)
+    This class is used within `ApplyConcatApply._simplify_down`.
 
-        keys = list(d)
-        j = 1
+    See Also
+    --------
+    ApplyConcatApply
+    """
+
+    _parameters = ["frame", "aca"]
+
+    @functools.cached_property
+    def _name(self):
+        return "chunk-" + tokenize(self.frame, self.aca._name)
+
+    @property
+    def aca(self):
+        return self.operand("aca").expr
+
+    @property
+    def chunk(self):
+        return self.aca.chunk
+
+    @property
+    def chunk_kwargs(self):
+        return self.aca.chunk_kwargs or {}
+
+    @functools.cached_property
+    def _meta(self):
+        meta = meta_nonempty(self.frame._meta)
+        chunk_kwargs = self.chunk_kwargs or {}
+        return make_meta(self.chunk(meta, **chunk_kwargs))
+
+    def _task(self, index: int):
+        args = [self._blockwise_arg(self.frame, index)]
+        chunk_kwargs = self.chunk_kwargs or {}
+        return (apply, self.chunk, args, chunk_kwargs)
```

Comment:
For ease of debugging task graphs I recommend the following:

```suggestion
        if self.chunk_kwargs:
            return (apply, self.chunk, args, self.chunk_kwargs)
        else:
            return (self.chunk,) + tuple(args)
```

Or maybe even bring args inside